### PR TITLE
Doc: Enhance and expand DLQ docs

### PR DESCRIPTION
**Pull request by karenzone**
**PREVIEW:** https://logstash_12959.docs-preview.app.elstc.co/guide/en/logstash/master/resiliency.html

## Release notes
Expand documentation for the dead letter queue (DLQ) to help users understand its purpose and management

## What does this PR do?
We have identified improvements for the DLQ (#10493), and those improvements are pending design, scoping, and prioritization. We can help users in the interim by better explaining the design intent and offering maintenance techniques to help them use DLQs more effectively. 

- Relates #12923
- Relates #10493

### CODE REVIEW

**Comment by karenzone** [comment]
@yaauie @acchen97  Thanks for the productive conversation on Wednesday. Here's a PR that captures what we discussed.
I noticed that the original docs issue mentioned tracking DLQ size on disk, and so I created a place in the doc for that info.  We can fill the gaps with another quick conversation, comments/suggestions in the PR, a thread in slack... whatever works best for you.

**Comment by karenzone** [comment]
@yaauie  Let's circle back around on this one when you have time, please.  <3

File: docs/static/dead-letter-queues.asciidoc [hunk 1]
```diff
@@ -250,3 +264,17 @@ output {
...
+
+ToDo: Add info on how to track size before it becomes a problem
+
+[[dlq-clear]]
+==== Clear the dead letter queue
+
+The dead letter queue cannot be cleared with the upstream pipeline running. 
+
+The dead letter queue is a directory of pages. 
+To clear it, stop the pipeline and delete location/<file-name>.
```
Comment (yaauie) [inline docs/static/dead-letter-queues.asciidoc:278]: `${path.data}/dead_letter_queue/${pipeline_id}`

File: docs/static/dead-letter-queues.asciidoc [hunk 2]
```diff
@@ -250,3 +264,17 @@ output {
 <3> The clean event is sent to Elasticsearch, where it can be indexed because
 the mapping issue is resolved.
 
+[[dlq-size]]
+==== Track dead letter queue size
+
+ToDo: Add info on how to track size before it becomes a problem
```
Comment (yaauie) [inline docs/static/dead-letter-queues.asciidoc:270]: The size of the DLQ for each pipeline that has it enabled is available in the node stats API at the address:
`pipelines.${pipeline_id}.dead_letter_queue.queue_size_in_bytes`.

**Review by yaauie — COMMENTED** [review]
I'm sorry for missing this -- I've added some specifics to hopefully help you fill the gaps. 

In general, I think this improves the docs by better setting the tone and expectations.

**Comment by karenzone** [comment]
@yaauie Thanks for the comments! I've added them and would appreciate another review.

**Review by yaauie — APPROVED** [review]
LGTM ✨

**Pull request merged**